File: frontend/src/api/uploads.ts
/**
 * 通用附件上传
 */
import request from './request'

export interface UploadResult {
  filename: string
  path: string
  url: string
  size: number
}

export function uploadFile(file: File, category = 'contract_proof') {
  const form = new FormData()
  form.append('file', file)
  form.append('category', category)
  return request.post<UploadResult>('/uploads', form, {
    timeout: 60000,
  })
}

File: backend/app/api/v1/uploads.py
"""通用文件上传 API。"""
from typing import Annotated

from fastapi import APIRouter, Depends, File, Form, UploadFile

from app.api.deps import PermissionChecker
from app.models.user import User
from app.services import uploads as upload_service

router = APIRouter(prefix="/uploads", tags=["文件上传"])


@router.post("", summary="上传附件")
async def upload_file(
    file: Annotated[UploadFile, File(...)],
    current_user: Annotated[
        User, Depends(PermissionChecker(["contract:view", "project:view"], any_of=True))
    ],
    category: Annotated[str, Form()] = "contract_proof",
) -> dict:
    _ = current_user
    return upload_service.save_upload(file, category=category)
